### ISSUE TITLE
Feature Request: Please support default that picks up value from environment variable

### ISSUE THREAD

**Issue by lenkite** [docopt/docopt.cpp#127]
This is a popular convention now being followed by several CLI tools that pick up default values (like tokens or passwords) from environment variables if not specified at the command line. Maybe something like 
`-t  <token>   api token  [default:env:NAVAL_FATE_TOKEN]`

But the author might have a better way to do this.

**Comment by Kerndog73** [comment]
For something like this, you might want to request it [upstream](https://github.com/docopt/docopt)

**Comment by AntonioCS** [comment]
@Kerndog73 does the c++ version only have things that the python version has?

**Comment by jaredgrubb** [comment]
Yes, we've tried to keep it with the same features as the Python version (and by extension, all the other ports).
